crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -96,20 +132,38 @@ impl RoomList {
     /// list entry's updates.
     pub fn entries(
         &self,
-    ) -> (Vector<RoomListEntry>, impl Stream<Item = VectorDiff<RoomListEntry>>) {
-        self.sliding_sync_list.room_list_stream()
+    ) -> (Vector<RoomListEntry>, impl Stream<Item = Vec<VectorDiff<RoomListEntry>>>) {
```

Comment:
How about not changing this method, and instead exposing a separate method to subscribe to the `entries_batch_drainer` (something like `subscribe_batch_processed(&self) -> eyeball::Subscriber<()>`)? That would mean

- the FFI layer can plug the two together, and the `eyeball_im::VectorDiff<_>` to `TimelineDiff` conversion can happen before batching (slightly more efficient)
- test code doesn't need to be changed

Reply:
I don't want to go that path because:

1. I prefer to get a very simple and small API so that we take full responsability for performance and quality
2. The FFI code is not tested, and I prefer this FFI part to be as small and thin as possible, at best to only be a “forwarding/wrapper API”
3. Test code didn't change that much, it was a small update on the macro: instead of consuming a stream (`Stream<Item = VectorDiff<_>>`), it consumes an iterator, which is basically the same! (`Vec<VectorDiff<_>>`).

Reply:
But is the batching something all consumers of the API want? As far as I understood it was really only done to minimize FFI calls? That concern doesn't apply for people who write code that consumes the Room List from Rust. An alternative could be to expose both a "regular", and a batched stream.

Reply:
I believe having a batch subscriber is the correct API here, as I don't see why someone wouldn't want this.

At worst, we could provide 4 methods instead of 2:

1. `entries` (already present)
2. `filtered_entries` (already present)
3. `batched_entries` (new)
4. `batched_filtered_entries` (new)

That way, we can test them on our side, and FFI can use whatever feels the best.

But I find it unnecessarily and very confusing, don't you think?

Reply:
I guess the batched one is almost identical in performance and not really harder to use. If you insist, let's merge as-is.